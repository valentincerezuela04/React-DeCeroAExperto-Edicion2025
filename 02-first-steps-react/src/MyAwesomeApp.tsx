import type { CSSProperties } from "react";

const firstName: string = 'valentin';
const lastName: string = 'cerezuela';

const favoriteGmaes: string[] = ['The Last of us', 'LOL', 'Elden Ring'];
const isActive = false;

const address = {
    zipCode: 'abc213',
    country: 'Canada',
};


const myStyles: CSSProperties = { //why CSSProperties ?? without this property we dont have autocomplete ,because visualStudioCode detected like a regular object
    backgroundColor: '#fafafa',
    borderRadius: 1,
    padding: 10

}

export const MyAwesomeApp = () => {
    return (
        <>
            <div data-testid="test-div">
                <h1 data-testid="first-name-title">{firstName}</h1>
                <h2>{lastName}</h2>
                <p>{favoriteGmaes.join(' ,')}</p>
                <p>{2 + 2}</p>
                <h1>{isActive ? 'active' : 'Not active'}</h1>

                <p style={myStyles}>
                    {JSON.stringify(address)}</p>
                <p>Hoal com o es que e</p>

            </div>
        </>
    );
};