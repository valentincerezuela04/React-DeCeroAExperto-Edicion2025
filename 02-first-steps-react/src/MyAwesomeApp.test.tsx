import { describe, test, expect } from "vitest";
import { render, screen } from '@testing-library/react';

import { MyAwesomeApp } from "./MyAwesomeApp";

describe('MyAwesomeApp', () => {
  test('should render firstName and lastName', () => {
    render(<MyAwesomeApp />);
    //console.log(container.innerHTML);
    //screen.debug(); //render all screen content of the component

    // const h1 = container.querySelector('h1');
    // const h2 = container.querySelector('h2');

    // expect(h1?.innerHTML).toContain('valentin');
    // expect(h2?.innerHTML).toContain('cerezuela');
    screen.debug();

    const h1 = screen.getByTestId('first-name-title');
    expect(h1.innerHTML).toContain('valentin');


  });
});
