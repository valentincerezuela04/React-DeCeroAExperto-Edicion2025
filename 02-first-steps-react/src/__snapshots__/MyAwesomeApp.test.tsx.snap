// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match snapshot 1`] = `
<div>
  <div
    data-testid="test-div"
  >
    <h1
      data-testid="first-name-title"
    >
      valentin
    </h1>
    <h2>
      cerezuela
    </h2>
    <p>
      The Last of us ,LOL ,Elden Ring
    </p>
    <p>
      4
    </p>
    <h1>
      Not active
    </h1>
    <p
      style="background-color: rgb(250, 250, 250); border-radius: 1px; padding: 10px;"
    >
      {"zipCode":"abc213","country":"Canada"}
    </p>
    <p>
      Hoal com o es que e
    </p>
  </div>
</div>
`;

exports[`MyAwesomeApp > should match snapshot div 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="test-div"
      >
        <h1
          data-testid="first-name-title"
        >
          valentin
        </h1>
        <h2>
          cerezuela
        </h2>
        <p>
          The Last of us ,LOL ,Elden Ring
        </p>
        <p>
          4
        </p>
        <h1>
          Not active
        </h1>
        <p
          style="background-color: rgb(250, 250, 250); border-radius: 1px; padding: 10px;"
        >
          {"zipCode":"abc213","country":"Canada"}
        </p>
        <p>
          Hoal com o es que e
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="test-div"
    >
      <h1
        data-testid="first-name-title"
      >
        valentin
      </h1>
      <h2>
        cerezuela
      </h2>
      <p>
        The Last of us ,LOL ,Elden Ring
      </p>
      <p>
        4
      </p>
      <h1>
        Not active
      </h1>
      <p
        style="background-color: rgb(250, 250, 250); border-radius: 1px; padding: 10px;"
      >
        {"zipCode":"abc213","country":"Canada"}
      </p>
      <p>
        Hoal com o es que e
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`MyAwesomeApp > should match snapshot div v2 1`] = `
<div
  data-testid="test-div"
>
  <h1
    data-testid="first-name-title"
  >
    valentin
  </h1>
  <h2>
    cerezuela
  </h2>
  <p>
    The Last of us ,LOL ,Elden Ring
  </p>
  <p>
    4
  </p>
  <h1>
    Not active
  </h1>
  <p
    style="background-color: rgb(250, 250, 250); border-radius: 1px; padding: 10px;"
  >
    {"zipCode":"abc213","country":"Canada"}
  </p>
  <p>
    Hoal com o es que e
  </p>
</div>
`;
